Assets/Scripts/Drone Scripts/Drone.cs

```diff
@@ -4,16 +4,20 @@
     using System.Collections.Generic;
     using UnityEngine;
     using ROSBridgeLib.interface_msgs;
+    using UnityEngine.UI;
 
     public class Drone
     {
 
         public GameObject gameObjectPointer; // This is the related game object
+        public DroneProperties droneProperties;
         public char id; // This is the identifier of the drone in the dronesDict and across the ROSBridge
         public bool selected;
 
         public ArrayList waypoints; // All waypoints held by the drone
-        public ArrayList waypointsOrder; // Keeps track of the order in which waypoints were created for the undo function
```

Comment:
Maybe add a comment specifying the idff between waypoints, waypointsorder and waypoints dict. And what's the dict holding? Strings for what?

Reply:
@apollovision

Reply:
Um, right now there are just two arrays, waypoints and deletedWaypoints. I can't seem to find the waypointsDict?

Reply:
I removed it :)